### PR TITLE
User following

### PR DESCRIPTION
**Pull request by melatran**
-Github following and Github followers story
-Refactored API consumption files to make skinny controllers
-Created models for repo, followers, following
-Tests passed

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
-      response = conn.get("/user/repos?access_token=#{current_user.github_token}")
-      
-      json = JSON.parse(response.body, symbolize_names: true)
-      @repos = json[0..4].map do |user_data|
-        Repo.new(user_data)   
-      end 
-    end  
+      search_results = SearchResults.new
+      @repos = search_results.repos
+      @followers = search_results.followers
```
Comment (sagemlee) [inline app/controllers/users_controller.rb:7]: This looks so much cleaner. Awesome job refactoring.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,22 +1,25 @@
...
+      @followings = search_results.followings
```
Comment (foleymichelle9) [inline app/controllers/users_controller.rb:8]: This cleans up our controller nicely! Nice refactor Melanie :)

File: spec/features/user/user_can_see_followers_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+describe 'User Dashboard' do
+  it 'displays users repos' do
+    user = create(:user, github_token: ENV["GITHUB_ACCESS_TOKEN"])
+
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)
+
+    visit dashboard_path
+
+    within('.github') do
+      expect(page).to have_css(".follower", count: 4)
```
Comment (melatran) [inline spec/features/user/user_can_see_followers_spec.rb:12]: We probably need to stub this so the followers count

File: app/poros/search_results.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    end
+  end
+
+  def followings
+    json = GithubService.new.list_followings
+    @followings = json.map do |user_data|
+      Following.new(user_data)
+    end
+  end
+end
```
Comment (foleymichelle9) [inline app/poros/search_results.rb:23]: Nice this along with our controller is such a great refactor!

**Pull request merged**